### ISSUE TITLE
WireMod Integration

### ISSUE THREAD

**Issue by ThePademelon** [ThePademelon/GModCryoChamber#25]
Integration instructions from Discord user, @Divran:

> alright, to add inputs you use WireLib.CreateInputs(ent, {"A","B [VECTOR]", "C [STRING]"}). same for outputs WireLib.CreateOutputs(ent, table). Normally, the entity will create these on itself in its initialize function, so you'd use self in place of ent
> next youre gonna want to inherit from the base wire entity https://github.com/wiremod/wire/blob/master/lua/entities/base_wire_entity.lua
> so like this 
> if WireLib then
>     DEFINE_BASECLASS("base_wire_entity")
>  else
>     DEFINE_BASECLASS("base_gmodentity")
> end

@bigdogmat however recommends setting the base class like so:

> ENT.Base = "base_wire_entity"

**Comment by bigdogmat** [comment]
Just to put what we found out in discord in a neat little format,

```lua
ENT.Base = "base_wire_entity"
```

^ will set the `BaseClass` variable and will work fine.


```lua
DEFINE_BASECLASS("base_wire_entity")
```

^ Is a literal hack in the parser that translates into

```lua
local BaseClass = baseclass.Get("base_wire_entity")
```

Reason this sets the `BaseClass` index on the ENT table is because `baseclass.Get` does `ENT.Base = name`, and so it's the same as the first solution, just more hacky.

**Comment by Divran** [comment]
The end result is identical for `DEFINE_BASECLASS` vs `ENT.Base`. `DEFINE_BASECLASS` just does more unecessary stuff (which don't matter much because they're only done once during server load). It also appears that the gmod developers themseleves use `DEFINE_BASECLASS` in the normal gmod entities, such as sent_ball.

TLDR: it doesn't matter, do what you want. You're gonna need that if statement in either case though, because trying to inherit from a class that doesn't exist would be bad.

**Comment by ThePademelon** [comment]
Thanks for the help guys.
